lib/msf/core/post/windows/version.rb

```diff
@@ -0,0 +1,67 @@
+# -*- coding: binary -*-
+
+module Msf::Post::Windows::Version
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_railgun_api
+            ]
+          }
+        }
+      )
+    )
+  end
+
+  def get_version_info
+    if session.type == 'meterpreter'
+      result = session.railgun.ntdll.RtlGetVersion(input_os_version_info_ex)
+      os_version_info_ex = unpack_version_info(result['VersionInformation'])
+      major = os_version_info_ex[1]
+      minor = os_version_info_ex[2]
+      build = os_version_info_ex[3]
+      service_pack = os_version_info_ex[6]
+      product_type = os_version_info_ex[9]
+
+      Msf::WindowsVersion.new(major, minor, build, service_pack, product_type)
+    else
```

Comment:
Unsure of best approach here, but simple stopgap would be to raise:

```suggestion
    else
      raise 'Unable to retrieve Windows version'
```